docs/apache-airflow-providers-google/connections/gcp.rst

```diff
@@ -160,10 +160,13 @@ access token, which will allow to act on its behalf using its permissions. ``imp
 does not even need to have a generated key.
 
 .. warning::
-  :class:`~airflow.providers.google.cloud.operators.kubernetes_engine.GKEStartPodOperator`,
   :class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreateJavaJobOperator` and
   :class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreatePythonJobOperator`
-  do not support direct impersonation as of now.
+  do not support direct impersonation as of now. Both are deprecated and new operators
+  :class:`~airflow.providers.apache.beam.operators.beam.BeamRunJavaPipelineOperator` and
```

Comment:
Did you check it? I think we are still missing the piece of code that will pass impersonation details to the Apache Beam subprocess. For now, only the monitoring process supports, but that might not be enough.

Reply:
@mik-laj based on the documentation: to the both operators we can pass [DataflowConfiguration](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_api/airflow/providers/google/cloud/operators/dataflow/index.html) where we can specify `impersonation_chain` parameter and we don't have any information that we support only monitoring process.

Reply:
[GoogleBaseHook.provide_authorized_gcloud](https://github.com/apache/airflow/blob/6e43137cac4f9adedcb1622c7178862a051ecd30/airflow/providers/google/common/hooks/base_google.py#L484) is used to pass credentials from to Apache Beam process. And this method does not consider impersonation chain as each application have a different way to pass the impersonation chain

I think this option was added by mistake in [this PR](https://github.com/apache/airflow/pull/12814/files) and has never been tested.

Reply:
@mik-laj I assumed that if we using that configuration and we don't have any warnings or information then it used inside the operators. I didn't run beam operators to verify that works or not. I only verified GKEStartPodOperator.

in the `provide_authorization_gcloud` we only set up the gcloud connection and we don't need there `--impersonate-service-account`, this parameter is only needed in the execution process. Authorization process still should use the orginal SA, please let me know if I'm wrong.

Reply:
We should use the same credentials in Beam operators and Kubernetes Engine Operator for submiting and monitoring tasks. I do not see the reason why the pod/beam job should be created with the use of one credential and then monitored with the use of another credentials. This likely won't work as each credential has a different permission scope and a task created by one account with one credential doesn't need to access the other account's data.

If we do not have any warnings and notifications, this is even more problematic because the user is not aware that we have not used the credentials selected by the user.

Reply:
@mik-laj So if now we support impersonation only in the monitoring tasks we should adjust the submitting tasks, to use impersonation, because from my perspective now this is confused for users.

Reply:
Yes. Exactly. Therefore, we cannot recommend new operators in the documentation, because this impersonation does not fully work.

Reply:
@mik-laj great that we are on the same page, I will remove recommendation in the documentation and I will prepare new PR to support impersonation in the Beam operators for Data Flow execution.